langchain/vectorstores/mongodb_atlas.py

```diff
@@ -0,0 +1,239 @@
+        self._index_name = index_name
+        self._text_key = text_key
+        self._embedding_key = embedding_key
+
+    @classmethod
+    def from_connection_string(
+        cls,
+        connection_string: str,
+        namespace: str,
+        embedding: Embeddings,
+        **kwargs: Any,
+    ) -> MongoDBAtlasVectorSearch:
+        try:
+            from pymongo import MongoClient
+        except ImportError:
+            raise ImportError(
+                "Could not import pymongo, please install it with "
+                "`pip install pymongo`."
+            )
+        client = MongoClient(connection_string)
+        return cls(client, namespace, embedding, **kwargs)
+
+    def add_texts(
+        self,
+        texts: Iterable[str],
+        metadatas: Optional[List[dict]] = None,
+        **kwargs: Any,
+    ) -> List:
+        """Run more texts through the embeddings and add to the vectorstore.
+
+        Args:
+            texts: Iterable of strings to add to the vectorstore.
+            metadatas: Optional list of metadatas associated with the texts.
+
+        Returns:
+            List of ids from adding the texts into the vectorstore.
+        """
+        # Embed and create the documents
+        docs = []
+        for i, text in enumerate(texts):
+            metadata = metadatas[i] if metadatas else {}
+            embedding = self._embedding.embed_documents([text])[0]
+            doc = {self._text_key: text, self._embedding_key: embedding, **metadata}
+            docs.append(doc)
+
+        # insert in MongoDB Atlas
+        insert_result = self._collection.insert_many(docs)
+        return insert_result.inserted_ids
+
+    def similarity_search_with_score(
+        self,
+        query: str,
+        *,
+        k: int = 4,
+        pre_filter: Optional[dict] = None,
+        post_filter_pipeline: Optional[List[Dict]] = None,
+    ) -> List[Tuple[Document, float]]:
+        """Return MongoDB documents most similar to query, along with scores.
+
+        Use the knnBeta Operator available in MongoDB Atlas Search
+        This feature is in early access and available only for evaluation purposes, to
+        validate functionality, and to gather feedback from a small closed group of
+        early access users. It is not recommended for production deployments as we
+        may introduce breaking changes.
+        For more: https://www.mongodb.com/docs/atlas/atlas-search/knn-beta
+
+        Args:
+            query: Text to look up documents similar to.
+            k: Optional Number of Documents to return. Defaults to 4.
+            pre_filter: Optional Dictionary of argument(s) to prefilter on document
+                fields.
+            post_filter_pipeline: Optional Pipeline of MongoDB aggregation stages
+                following the knnBeta search.
+
+        Returns:
+            List of Documents most similar to the query and score for each
+        """
+        pipeline = [
```

Comment:
@P-E-B, it appears we are only implementing support for the "Basic" and "Filter" scenario. 

https://www.mongodb.com/docs/atlas/atlas-search/knn-beta/#examples

Could this segment be adjusted to include support for "Compound" scenarios as well?

Reply:
That's a good point. While it's more efficient to filter completely on Lucene side, the purpose of this method is to retrieve k documents that are similar in a semantic way. Using compound would jeopardize this (using must would result in post filtering) and possibly return less that k documents. I'll maybe code a new method that will make this feature available if you deem it's relevant